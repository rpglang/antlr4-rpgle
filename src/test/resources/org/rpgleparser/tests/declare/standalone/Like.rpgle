        // Define using the LIKE keyword 
        DCL-S cust_index LIKE(index);

        //Specify length adjustment with LIKE keyword 
        DCL-S big_index LIKE(index : +6);