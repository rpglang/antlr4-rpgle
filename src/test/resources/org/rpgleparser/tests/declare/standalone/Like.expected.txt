/*===TOKENS===*/
HIDDEN:COMMENTS          <   //>
HIDDEN:COMMENTS_TEXT     <Define using the LIKE keyword>
DS_Standalone            <DCL-S>
ID                       <cust_index>
KEYWORD_LIKE             <LIKE>
'('                      <(>
ID                       <index>
')'                      <)>
FREE_SEMI                <;>
HIDDEN:COMMENTS          <   //>
HIDDEN:COMMENTS_TEXT     <Specify length adjustment with LIKE keyword>
DS_Standalone            <DCL-S>
ID                       <big_index>
KEYWORD_LIKE             <LIKE>
'('                      <(>
ID                       <index>
COLON                    <:>
'+'                      <+>
NUMBER                   <6>
')'                      <)>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword cust_index)))
      (keyword
        (keyword_like
          LIKE
          (
          (simpleExpression (identifier (free_identifier (idOrKeyword index))))
          )
        )
      )
      ;
    )
  )
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword big_index)))
      (keyword
        (keyword_like
          LIKE
          (
          (simpleExpression (identifier (free_identifier (idOrKeyword index))))
          :
          (like_lengthAdjustment (sign +) (number 6))
          )
        )
      )
      ;
    )
  )
)
/*======*/